### ISSUE TITLE
Explain coordinate system in P5.js

### ISSUE THREAD

**Issue by benjiallen** [benjiallen/coderdojochi-js#1]
## Learning outcome

In school when you're drawing graphs with an x and y axis (0, 0) is at the bottom left. In computer graphics it's a little bit different. (0,0) is in the top left corner. This can be a little confusing at first but once you get used to it it's easy!
## To do
- Need some way of explaining this initial concept. Could just be a drawing on a white board
- Need lots of examples. Let's take a 600x600 canvas and show lots of different points.
  - Again, could be a drawing on a whiteboard

**Comment by pjsier** [comment]
Is there a way we could have a grid display on the screen for them? I think p5 is our best option, but I'm worried about the grid being really abstract. Maybe we could have some JS for an interactive grid (numbers highlighted on hover) included in the back? Not sure if that's overkill, but that'd be my main concern.

**Comment by benjiallen** [comment]
Thanks for putting this together Pat.

I really like the idea of explaining the coordinate system with some form of visual. I would expect kids to be more comfortable with grid paper and doing this in p5 is pretty simple.

A tool tip to show the coordinate will also be super helpful.

Perhaps we can take this idea one step further and build a toggle for debugging which turns the grid and coordinate tooltip on and off. Kids can quickly get their bearings and work out where there next point needs to go. What do you think?

**Comment by benjiallen** [comment]
![grid-example](https://cloud.githubusercontent.com/assets/16546334/13939906/f0194f10-efa6-11e5-870b-f905314c804a.png)

@pjsier here is a mockup explaining my comment above.

**Comment by pjsier** [comment]
I think it could work, my only worry is overcomplicating things for the students on their `setup()` and `draw()` functions. 

We'd have to have some kind of closure around it, and I think it might be possible, but given that it seems p5 relies on those two functions, I'd be worried about playing with them too much to the point where kids can't do the same thing after they leave class. I think it's something we can talk about though.

**Comment by benjiallen** [comment]
Good point and agreed. I think this is only an option if we can keep the setup and draw functions "clean".

**Comment by benjiallen** [comment]
@pjsier failing a really nice JS helper, I've put together an easy exercise for getting kids warmed up to the idea of how x and y coordinates work in computer graphics.

![coordinate-exercises](https://cloud.githubusercontent.com/assets/16546334/14588735/165d8138-0496-11e6-839d-488f51cc73a3.png)

**Comment by pjsier** [comment]
@benjiallen great! Are you thinking we'd need to change the coordinates page, or is this just more of a way of explaining it? I think it could work for sure, and we'd definitely need something to introduce them to it

**Comment by benjiallen** [comment]
@pjsier - i thought this would be a simple exercise to help explain the coordinate system. I was thinking we could print this out and just have students do a pen and paper exercise.

Also, i've taken your grid example and added to it. It's closer to [the mockup](https://cloud.githubusercontent.com/assets/16546334/13939906/f0194f10-efa6-11e5-870b-f905314c804a.png) and it keeps the draw function clean. You do have to add a little bit of code to `setup` but I try to minimize it.

Check it out http://benjiallen.github.io/coderdojochi-js/utilities/grid-toggle/index.html

**Comment by pjsier** [comment]
@benjiallen I think that works great! I tested out moving all of the grid-related functions to a separate file, and all I needed to do to bring the grid into any other example was `canvas.parent("container")` and `gridSetup()`. 

I think this is definitely the best way of doing this to include the grid.

**Comment by benjiallen** [comment]
Great. We'll keep this in our back pocket. If students need it, we'll be ready :)
